deps/build.jl

```diff
@@ -0,0 +1,91 @@
+# Builds a GAP, and attaches JuliaInterface
+
+import Pkg
+
+julia_folder_path = abspath(joinpath(@__DIR__,".."))
+
+gap_root = abspath(joinpath(julia_folder_path,"gap"))
+install_gap = true
+
+if haskey(ENV,"GAP_ROOT")
+    gap_root = ENV["GAP_ROOT"]
+    install_gap = false
+end
+
+## Find julia binary
+
+julia_binary = missing
+
+if haskey(ENV,"JULIA_BINARY")
+    julia_binary = ENV["JULIA_BINARY"]
+else
+    julia_binary = Sys.BINDIR
+end
+
+## Install GAP
+if install_gap
+    cd(julia_folder_path)
+    run(`git clone https://github.com/gap-system/gap`)
+    cd(gap_root)
+    run(`./autogen.sh`)
+    run(`./configure --with-gc=julia --with-julia=$(julia_binary)`)
+    run(`make`)
+    if ! haskey(ENV,"GAP_INSTALL_PACKAGES") || ENV["GAP_INSTALL_PACKAGES"] == "yes" 
+        run(`make bootstrap-pkg-full`)
+        cd("pkg")
+        run(`../bin/BuildPackages.sh`)
+    elseif haskey(ENV,"GAP_INSTALL_PACKAGES") && ENV["GAP_INSTALL_PACKAGES"] == "minimal"
+        run(`make bootstrap-pkg-minimal`)
+    else
+        run(`mkdir pkg`)
```

Comment:
It might be that people want to not download or compile any packages, as they might want to symlink a custom `pkg` folder into this GAP. Do you think this is unnecessary to create the folder?